.github/workflows/publish.sh

```diff
@@ -0,0 +1,33 @@
+#/bin/sh
+# -- Create the rockspec that will be published to LuaRocks
+
+# LuaRocks works by accepting a .rockspec file, then mirroring the code of that tag from github
+# This script generates a versioned .rockspec
+
+# The tag pushed to github
+TAG_VERSION=${GITHUB_REF#refs/*/}
+
+ROCKSPEC_DIR=./.rockspec
+
+EXISTING_ROCKSPEC_PATH=$(find . -name "snowplowtracker-$TAG_VERSION-1.rockspec")
+[ -n "$EXISTING_ROCKSPEC_PATH" ] && echo "A rockspec of version $TAG_VERSION already exists." && exit 1
+
+NEW_ROCKSPEC_PATH="$ROCKSPEC_DIR"/snowplowtracker-"$TAG_VERSION"-1.rockspec
+cp ./snowplowtracker-scm-1.rockspec "$NEW_ROCKSPEC_PATH"
+
+# Replace the version number in the copied scm-1 with the release tag
+sed -i 's|version = \"scm-1\"|version = \"'"$TAG_VERSION"'-1\"|g' "$NEW_ROCKSPEC_PATH"
+
+# Set the github tag in the new rockspec
+sed -i 's|SET_BY_GITHUB_ACTIONS_DO_NOT_MANUALLY_SET|'"$TAG_VERSION"'|g' "$NEW_ROCKSPEC_PATH"
+
+git add "$NEW_ROCKSPEC_PATH"
+git commit --no-verify -m "Add rockspec version $TAG_VERSION"
```

Comment:
Confused about this – so the script replaces the scm placeholder with a version and then makes a git commit? I don't see a git push, is this commit thrown away after publishing to luarocks?